src/components/cc-invoice/cc-invoice.js

```diff
@@ -32,49 +32,64 @@ export class CcInvoice extends LitElement {
 
   static get properties () {
     return {
-      error: { type: Boolean },
       invoice: { type: Object },
-      number: { type: String },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {boolean} Sets a loading error state. */
-    this.error = false;
-
-    /** @type {Invoice|null} Sets the invoice. */
-    this.invoice = null;
+    /** @type {InvoiceState} Sets the invoice state. */
+    this.invoice = { state: 'loading' };
+  }
 
-    /** @type {string|null} Sets the invoice number. */
-    this.number = null;
+  _getInvoiceNumber () {
+    if (this.invoice.state === 'loading') {
+      return this.invoice.number;
+    }
+    if (this.invoice.state === 'error') {
+      return this.invoice.number;
+    }
+    if (this.invoice.state === 'loaded') {
+      return this.invoice.invoice.number;
+    }
+    return null;
   }
 
   render () {
 
-    const skeleton = (this.invoice == null);
-    const invoice = skeleton ? SKELETON_INVOICE : this.invoice;
-
-    const number = this.number ?? SKELETON_INVOICE.number;
-    const date = invoice.emissionDate;
-    const amount = invoice.total.amount;
+    const invoiceNumber = this._getInvoiceNumber();
+    const skeletonNumber = invoiceNumber == null;
+    const number = skeletonNumber ? SKELETON_INVOICE.number : invoiceNumber;
+    const skeleton = skeletonNumber || (this.invoice.state === 'loading');
+    const invoice = skeleton ? SKELETON_INVOICE : this.invoice.invoice;
+    const title = `${i18n('cc-invoice.title')} ${number}`;
 
     return html`
-      <cc-block .icon=${iconFile} class=${classMap({ 'has-errors': this.error })}>
-        <div slot="title">${i18n('cc-invoice.title')} ${number}</div>
-        ${!this.error ? html`
-          <div slot="button">${ccLink(invoice.downloadUrl, i18n('cc-invoice.download-pdf'), skeleton)}</div>
-          <div class="info"><em class=${classMap({ skeleton })}>${i18n('cc-invoice.info', { date, amount })}</em></div>
-          <cc-html-frame class="frame" ?loading="${skeleton}" iframe-title="${i18n('cc-invoice.title')} ${number}">
-            ${!skeleton ? unsafeHTML(`
-              <template>${this.invoice.invoiceHtml}</template>
-            `) : ''}
-          </cc-html-frame>
-        ` : ''}
-        ${this.error ? html`
+      <cc-block .icon=${iconFile} class=${classMap({ 'has-errors': this.invoice.state === 'error' })}>
+        <div slot="title">
+          ${skeletonNumber ? html`
+            ${i18n('cc-invoice.title')} <span class="skeleton">${number}</span>
+          ` : title}
+        </div>
```

Comment:
**suggestion:** 
```suggestion
        <div slot="title">
          ${i18n('cc-invoice.title')}
          <span class=${classMap({ skeleton: skeletonNumber })}>
            ${number}
          </span>
        </div>
```

Reply:
Thank you for your suggestion, but we need the `title` for the iframe title. So in fact, this will not simply things so much.